docs/contributing/style-guideline.md

```diff
@@ -96,3 +96,22 @@ Method parameters should never be declared `final`.
 
 Local variables should only be declared `final` if they are not initialized inline
 (declaring these vars `final` can help prevent accidental double-initialization).
+
+### `@Nullable` annotation usage
+
+[Note: this section is aspirational, as opposed to a reflection of the current codebase]
+
+All parameters and fields which can be `null` should be annotated with `@Nullable`
+(specifically `org.checkerframework.checker.nullness.qual.Nullable`, which is included by the
+`otel.java-conventions` gradle plugin as a `compileOnly` dependency).
+
+The default is assumed to be non-null, so there is no need to use `@NonNull`.
```

Comment:
This is not enough for Idea to highlight when `null` is passed to non-null parameter.

Reply:
You need `@ParametersAreNonnullByDefault` in `package-info.java` for that. Should we start adding those? At the very least to the *-api modules.

Reply:
First I would like to make it clear: does this mean that we don't need to defensively check passed attributes for null? Because SDK is full of those, where they check if `ParametersAreNonnullByDefault` parameter is actually not null.

Reply:
@iNikem PTAL I think I addressed your questions